sockeye/config.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not
+# use this file except in compliance with the License. A copy of the License
+# is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import copy
+
+import yaml
+
+
+class Config(yaml.YAMLObject):
+    """
+    Base configuration object that supports freezing of members and JSON (de-)serialization.
+    Actual Configuration should subclass this object.
+    """
+    yaml_tag = u'!Config'
+
+    def __init__(self):
+        self._frozen = False
+
+    def __setattr__(self, key, value):
+        if hasattr(self, '_frozen') and self._frozen:
```

Comment:
If we use __add_frozen() consitently (see constructor), we may skip the hasattr() test.